src/shell.ts

```diff
@@ -316,7 +316,7 @@ export class Shell {
             this.watchdogResolve = resolve
             this.startHourglass(timeout)
```

Comment:
Thanks for the PR!

the `startHourglass` is where the progress bar is printed and it also needs to be updated: receive the prefix and print it before the progress bar.